WasatchNET/SPISpectrometer.cs

```diff
@@ -470,25 +483,44 @@ public bool writeEEPROM(List<byte[]> pages)
             {
                 byte[] transmitData = pages[page];
 
+
+                logger.hexdump(transmitData, String.Format("preparing to write page {0}: ", page));
+
                 //we've had issues with writing EEPROM using SPI so the attempt here is to give extra time before telling the FPGA to write.
                 //a thread sleep might be more prudent, but this needs work in some way regardless
                 byte[] command = wrapCommand(WRITE_EEPROM_BUFFER, transmitData, STANDARD_PADDING * 4);
 
+                logger.hexdump(command, String.Format("write command for page {0}: ", page));
+
                 byte[] result = spi.readWrite(command);
 
+                logger.hexdump(result, "response: ");
+
                 //Thread.Sleep(100);
 
+                Thread.Sleep(2);
+
+                /*
                 transmitData = new byte[0];
-                command = wrapCommand(READ_EEPROM_BUFFER, transmitData, 100);
+                command = wrapCommand(READ_EEPROM_BUFFER, transmitData, 100); 
+                logger.hexdump(command, String.Format("buffer read back command for page {0}: ", page));
                 result = spi.readWrite(command);
 
+                logger.hexdump(result, "response: ");
+                */
+
                 //Thread.Sleep(100);
 
                 byte currPage = (byte)(0x80 | (uint)page);
 
                 transmitData = new byte[1] { currPage };
                 command = wrapCommand(EEPROM_OPS, transmitData, STANDARD_PADDING);
+                logger.hexdump(command, String.Format("commit to EEPROM command for page {0}: ", page));
                 result = spi.readWrite(command);
+
+                logger.hexdump(result, "response: ");
+
+                Thread.Sleep(2);

```

Comment:
Is this actually required?

Reply:
The sleep or the dump?

Reply:
It looks like it's there for the loop around